### ISSUE TITLE
How to import generated client when using CommonJS require syntax

### ISSUE THREAD

**Issue by nocin** [SAP/cloud-sdk-js#4339]
Hi Experts,

I feel stupid, but I cannot find the right way to import my generated client.
I successfully generated a client for the SucessFactors PLTUserManagement service ([PLTUserManagement.edmx](https://api.sap.com/api/PLTUserManagement/overview))

![grafik](https://github.com/SAP/cloud-sdk-js/assets/35531968/858988fd-9ada-4ce3-9043-28370cf63cae)

How do I access the User class, which is generated and exported in the index.ts file?

I tried different approaches, for example `const { User } = require('./external/PLTUserManagement')` but it is always undefined.

How do I even know what to import? For example [here](https://blogs.sap.com/2023/01/14/cap-handling-of-media-data-with-malware-scanning/), it is 'default-api'.

Can anyone help me out? 

Thanks & Regards
Nico

**Comment by deekshas8** [comment]
Hi @nocin ,
The 'default-api' in the example is an OpenAPI. For the OData service you would need to import the common service function and create a request as described [here](https://sap.github.io/cloud-sdk/docs/js/features/odata/execute-request#general-request-structure).

The syntax would be: `const { myEntityService } = require('./outputDir/my-service');`

Note: You should use the service function and not call the API constructor directly.

**Comment by nocin** [comment]
Hi @deekshas8,

sorry, but I still don't get it. How do I know the common service function?
When checking [here](https://api.sap.com/api/PLTUserManagement/cloud-sdk/JavaScript), it is done like this:

```
import { pltUserManagementService } from './generated/plt-user-management-service';

const { userApi } = pltUserManagementService();
const resultPromise = userApi.requestBuilder().getAll().top(5).execute({ destinationName: 'myDestinationName' });
```

So `pltUserManagementService ` is the service function?
And `plt-user-management-service` should be the generated folder?

I've just followed the guide again, but in my case the generated folder is PLTUserManagement, like in the screenshot in my initial post. I now tried:
`const { pltUserManagementService } = require('./external/plt-user-management-service');` leads to an error.
`const { pltUserManagementService } = require('./external/PLTUserManagement');` is still undefined.

I still cannot find out, what I'm doing wrong? Could you help once more? Thanks!

**Comment by deekshas8** [comment]
Hi @nocin ,

Sorry for the confusion, the example from the docs is not completely accurate. I'll fix that. In your case, I believe the service should be called `pltUserManagement`. Ideally the name of the service function is the directory name in camel casing. I will update our docs so its stated more explicitly. I thought we already did that. 

To be sure, you can also check the name of the exported function in `service.ts` file

**Comment by nocin** [comment]
Hi @deekshas8,

the exported function is indeed called `pltUserManagement` in the service.ts file.
![grafik](https://github.com/SAP/cloud-sdk-js/assets/35531968/ba3c9e9c-0797-4ce1-9ba3-37674e610cb2)

But it is still not working, as pltUserManagement is still undefined.
![grafik](https://github.com/SAP/cloud-sdk-js/assets/35531968/83d35262-0e60-4ed7-83fc-1e01416f9b98)
![grafik](https://github.com/SAP/cloud-sdk-js/assets/35531968/9e1c426e-c345-4f7e-918d-55b6a8e318d0)

Could you try to import the [PLTUserManagement.edmx](https://api.sap.com/api/PLTUserManagement/overview) yourself?
I have really no idea what I'm doing wrong anymore.

**Comment by mr-flannery** [comment]
Hi @nocin ,

I ran the generator on the file you linked and cannot find any problems. 

1. I installed `@sap-cloud-sdk/generator` version `3.9.0`.
2. I ran the following command:
    ```
    npx generate-odata-client --input generator-input/ --outputDir src/gen/ --skipValidation --clearOutputDir --transpile
    ```
3. Then ran the following file:
    ```js
    const { pltUserManagement } = require('./gen/PLTUserManagement')
    const apis = pltUserManagement()
    console.log(apis.userApi)
    ```

My node version is `v18.17.0`. What is your node version?

**Comment by nocin** [comment]
Hi @mr-flannery,

thanks to your example, I now got my issue. I was missing the `--transpile` parameter.

This was my command all the time:
`npx generate-odata-client --input srv/external/PLTUserManagement.edmx --outputDir srv/external/ --skipValidation`

When adding `--transpile`, it is working fine!
`npx generate-odata-client --input srv/external/PLTUserManagement.edmx --outputDir srv/external/ --skipValidation --transpile`

I just took the command from [here](https://api.sap.com/api/PLTUserManagement/cloud-sdk/JavaScript), without adjusting it. 
I expected, that importing a class from a .ts file should work, as I also saw this in many node packages, for example the buffer package. 

BR,
Nico
